packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -114,9 +110,50 @@ class TimeOfDay {
   @override
   int get hashCode => hashValues(hour, minute);
 
-  // TODO(ianh): Localize.
   @override
-  String toString() => '$hourOfPeriodLabel:$minuteLabel $periodLabel';
+  String toString() => '$hourLabel:$minuteLabel';
+}
+
+/// The list of time of day formats supported by this library encoded as ICU
+/// "Short Time" patterns.
+///
+/// The time picker invoked via [showTimePicker] has configurations optimized
+/// for these formats. Any other formats will default to [h_colon_mm_space_a].
+/// The time picker get the value from [MaterialLocalizations.timeOfDayFormat].
+abstract class TimeOfDayFormats {
+  // Patterns using HH.
+
+  /// One of the reusable time patterns and also the default format.
+  ///
+  /// This format is used as a fallback when the value of
+  /// [MaterialLocalizations.timeOfDayFormat] is not supported, i.e. not one of
+  /// the patterns listed in this class.
+  static const String HH_colon_mm = 'HH:mm';
```

Comment:
This one is labeled "One of the reusable time patterns" but the others "The 'HH.mm' pattern", which seems inconsistent.

Reply:
I decided to convert it to an enum. This gives us better type feedback and I don't have to make the patterns themselves public. It decouples the mapping of patterns to supported layouts (e.g. the enum does not contain the Thai-specific pattern, since we don't have a special layout for it). I made the enum public so I can use it as the return type of `MaterialLocalizations.timeOfDayFormat`. All the docs are on the enum.